### PR TITLE
Upgrade Bootstrap to 4.0.0

### PR DESCRIPTION
**Pull request by andrew**
screen shot 2017-05-11 at 9 34 06 pm

Looks like it's not the simplest upgrade, don't merge this pr but feel free to poke around and fork/update if you're feeling brave.

### CODE REVIEW

**Comment by andrew** [comment]
There's an actual upgrade guide here btw: https://v4-alpha.getbootstrap.com/migration/

**Comment by andrew** [comment]
Updated to 4.0.0.beta, looks slightly less broken than before: 
screen shot 2017-08-19 at 6 28 20 pm

**Comment by andrew** [comment]
screen shot 2017-08-19 at 9 10 43 pm

Fixed up the header (except on mobile)

**Comment by andrew** [comment]
Making more progress: 
screen shot 2017-08-19 at 9 35 06 pm

**Comment by samjacobclift** [comment]
I had a fair stab at this here https://github.com/samjacobclift/octobox/tree/bootstrap-4

**Comment by andrew** [comment]
@samjacobclift nice, want to open a fresh pr and replace this one? Got a screenshot?

**Comment by samjacobclift** [comment]
Let me see if I can get one going! should be able to rebase that branch and update it

**Comment by samjacobclift** [comment]
just resurrecting that PR might be easier to work off this branch and see how we go

File: Gemfile [hunk 1]
```diff
@@ -2,7 +2,7 @@ source 'https://rubygems.org'
 ruby '2.4.1'
 
 gem 'rails', '~> 5.1', '>= 5.1.3'
-gem 'bootstrap-sass'
+gem 'bootstrap', '~> 4.0.0.beta'
```
Comment (chrisarcand) [inline Gemfile]: TIL this is still beta, hah.

**Comment by maxibanki** [comment]
Hi, btw. Bootstrap 4 was official released, so that is isn't a beta anymore. :)

**Comment by andrew** [comment]
This PR isn't actually in as bad state as it looks, @samjacobclift made good progress in a separate PR, lots of screenshots over there: https://github.com/octobox/octobox/pull/436

**Comment by andrew** [comment]
Outstanding bug list:

- [ ] Label styles:

screen shot 2018-07-30 at 11 07 05

- [ ] Pagination position:

screen shot 2018-07-30 at 11 07 18

- [ ] active Menu highlight missing

screen shot 2018-07-30 at 11 07 51

- [ ] keyboard menu popup close button position and header size:

screen shot 2018-07-30 at 11 07 59

- [ ] sync button different size to filter buttons

screen shot 2018-07-30 at 11 08 17

- [ ] X button position

screen shot 2018-07-30 at 11 09 01

- [ ] broken API doc link in menu

screen shot 2018-07-30 at 11 09 25

- [ ] Broken API token buttons in settings page

screen shot 2018-07-30 at 11 10 02

- [ ] Settings page current git sha position:

screen shot 2018-07-30 at 11 10 11

- [ ] Logged out homepage alignment and text sizing:

screen shot 2018-07-30 at 11 10 52

**Comment by BenJam** [comment]
@andrew ready to take a look again when you're ready 👍

**Comment by BenJam** [comment]
- [ ] pagination not showing in sm devices
- [ ] scrolling limited on sm devices
- [ ] search bar behaviour different

File: app/assets/stylesheets/application.scss [hunk 2]
```diff
@@ -66,10 +66,10 @@ html {
       padding-left: 15px;
       overflow-y: hidden;
       height: 100%;
-      > .panel {
+      > .card {
         height: 100%;
         overflow-y: scroll;
-        > .panel-heading {
+        > .card-header {
```
Comment (BenJam) [inline app/assets/stylesheets/application.scss]: looks like these older styles aren't working in bs4 :/

**Comment by BenJam** [comment]
so.... close...

**Comment by andrew** [comment]
Couple more little bits:

- [ ] No hover states on sidebar menu items, bootstrap 3 example:

screen shot 2018-08-01 at 17 19 43

- [ ] Icons in filter buttons need a little spacing:

screen shot 2018-08-01 at 17 20 05

- [ ] Clickable target area on per-page menu is much smaller on this branch that bs3:

screen shot 2018-08-01 at 17 20 38

- [ ] Big space at the bottom of the table after the last item:

screen shot 2018-08-01 at 17 23 31

- [ ] Pagination disappears on xs-screens

Almost there!

**Comment by andrew** [comment]
One last one that might not be an easy fix, the icons looks significantly worse in Firefox than chrome/safari on a non-retina screen, could just be my setup though:
screen shot 2018-08-01 at 17 32 01

**Comment by andrew** [comment]
I think we're ready to go 🎉

**Comment by andrew** [comment]
Deploying this to https://octobox.io now 🚢

**Comment by RX14** [comment]
Before:



After:



The sidebar looks great, but the rest of the page seems to have been inflated to gigantic proportions.

I can fix it by adding `font-size: 14px` to the `:root`, but the text in the table header is then too small:

**Comment by andrew** [comment]
@RX14 would you be able to open that as a separate issue so it doesn't get lost in this closed thread?

**Pull request merged**